src/transformers/models/cpmant/modeling_cpmant.py

```diff
@@ -0,0 +1,1090 @@
+            dim_in=dim_model,
+            dim_ff=dim_ff,
+        )
+
+        if dropout_p is not None:
+            self.dropout = torch.nn.Dropout(dropout_p)
+        else:
+            self.dropout = None
+
+        self.w_out = nn.Linear(dim_ff, dim_model, bias=False)
+
+    def forward(self, x: torch.Tensor):
+        """
+        Args:
+            x (`torch.Tensor` of shape `(batch, seq_len, dim_in)`)
+        """  # noqa: E501
+        x = self.w_in(x)
+
+        if self.dropout is not None:
+            x = self.dropout(x)
+
+        x = self.w_out(x)
+
+        return x
+
+
+class CPMAntFFNBlock(nn.Module):
+    """The whole feed-forward block. A sequence of operation. Consists of layernorm, feed-forward and residual connection.
+
+    Args:
+        dim_model (int): Main dimension of modules in transformer blocks.
+        dim_ff (int): Dimension of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
+        eps (float, optional): The epsilon used by the layer normalization layers.
+        dropout_p (float, optional): Defaults to 0.
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        dim_model: int,
+        dim_ff: int,
+        eps: float = 1e-6,
+        dropout_p: Optional[float] = 0,
+    ):
+        super().__init__()
+
+        self.layernorm_before_ffn = CPMAntLayerNorm(
+            dim_model,
+            eps=eps,
+        )
+
+        self.ffn = CPMAntFeedForward(
+            dim_model,
+            dim_ff,
+            dropout_p=dropout_p,
+        )
+
+        if dropout_p:
+            self.dropout = torch.nn.Dropout(dropout_p)
+        else:
+            self.dropout = None
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+    ):
+        """
+        Args:
+            hidden_states (`torch.Tensor` of shape `(batch, len_seq, dim_model)`):
+                Hidden states before feed forward layer.
+        """  # noqa: E501
+        x = self.layernorm_before_ffn(hidden_states)
+        x = self.ffn(x)
+        if self.dropout is not None:
+            x = self.dropout(x)
+        hidden_states = hidden_states + x
+        return hidden_states
+
+
+class CPMAntTransformerBlock(nn.Module):
+    """The whole transformer block. A sequence of operation. Consists of self-attention block[, cross-attention block] and
+    feed-forward block.
+
+        Args:
+            dim_model (int): Main dimension of modules in transformer blocks.
+            dim_ff (int): Dimension of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
+            num_heads (int): Number of attention heads in the Transformer encoder.
+            dim_head (int): Dimension of attention heads for each attention layer in the Transformer encoder.
+            eps (float, optional): The epsilon used by the layer normalization layers.
+            dropout_p (float, optional): Defaults to 0.
+            use_cache (bool, optional): Whether to use cache.
```

Comment:
same comment as above, if you want to comment this class, consider using the `transformers` convention